pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaSSLUtils.java

```diff
@@ -0,0 +1,340 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyFactory;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * SSL utils class which helps in initialization of Kafka client SSL configuration. The class can install the
+ * provided server certificate enabling one-way SSL or it can install the server certificate and the
+ * client certificates enabling two-way SSL.
+ */
+public class KafkaSSLUtils {
+
+  private KafkaSSLUtils() {
+    // private on purpose
+  }
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaSSLUtils.class);
+  // Value constants
+  private static final String DEFAULT_CERTIFICATE_TYPE = "X.509";
+  private static final String DEFAULT_KEY_ALGORITHM = "RSA";
+  private static final String DEFAULT_KEYSTORE_TYPE = "PKCS12";
+  private static final String DEFAULT_SECURITY_PROTOCOL = "SSL";
+  private static final String DEFAULT_TRUSTSTORE_TYPE = "jks";
+  private static final String DEFAULT_SERVER_ALIAS = "ServerAlias";
+  private static final String DEFAULT_CLIENT_ALIAS = "ClientAlias";
+  // Key constants
+  private static final String SSL_TRUSTSTORE_LOCATION = "ssl.truststore.location";
+  private static final String SSL_TRUSTSTORE_PASSWORD = "ssl.truststore.password";
+  private static final String SECURITY_PROTOCOL = "security.protocol";
+  private static final String SSL_KEYSTORE_LOCATION = "ssl.keystore.location";
+  private static final String SSL_KEYSTORE_PASSWORD = "ssl.keystore.password";
+  private static final String SSL_KEY_PASSWORD = "ssl.key.password";
+  private static final String STREAM_KAFKA_SSL_SERVER_CERTIFICATE = "stream.kafka.ssl.server.certificate";
+  private static final String STREAM_KAFKA_SSL_CERTIFICATE_TYPE = "stream.kafka.ssl.certificate.type";
+  private static final String SSL_TRUSTSTORE_TYPE = "ssl.truststore.type";
+  private static final String STREAM_KAFKA_SSL_CLIENT_CERTIFICATE = "stream.kafka.ssl.client.certificate";
+  private static final String STREAM_KAFKA_SSL_CLIENT_KEY = "stream.kafka.ssl.client.key";
+  private static final String STREAM_KAFKA_SSL_CLIENT_KEY_ALGORITHM = "stream.kafka.ssl.client.key.algorithm";
+  private static final String SSL_KEYSTORE_TYPE = "ssl.keystore.type";
+
+  public static void initSSL(Properties consumerProps) {
+    // Check if one-way SSL is enabled. In this scenario, the client validates the server certificate.
+    String trustStoreLocation = consumerProps.getProperty(SSL_TRUSTSTORE_LOCATION);
+    String trustStorePassword = consumerProps.getProperty(SSL_TRUSTSTORE_PASSWORD);
+    if (StringUtils.isAnyEmpty(trustStoreLocation, trustStorePassword)) {
+      return;
```

Comment:
Can we add some info message here that we are skipping SSL initialization and its reason?

Reply:
done.